packages/playwright-core/src/server/network.ts

```diff
@@ -173,8 +173,22 @@ export class Request extends SdkObject {
     return this._overrides?.method || this._method;
   }
 
+  async body(): Promise<Buffer | null> {
+    if (this._overrides?.postData)
+      return this._overrides?.postData;
+    if (typeof this._postData === 'function')
```

Comment:
Let's keep `postDataBuffer` as is and introduce a separate `bodyCallback` which will be non-null only in Bidi for now. This method should become something like this:

```js
    if (this._overrides?.postData)
      return this._overrides?.postData;
    if (this._bodyCallback)
      return await this._bodyCallback();
    return this._postData;
```